styleguide/STYLEGUIDE.md

```diff
@@ -1,50 +1,1090 @@
+let user = try await getUser(
+  for: userID,
+  on: connection)
+```
 
-If you separate a function definition or call on to separate lines (because of the line length rule, perhaps) then _each_ parameter should be on a separate line.  Xcode should automatically align them for you.
+**Not Preferred**:
+```swift
+let user = try await getUser(
+  for: userID,
+  on: connection
+)
+```
 
-This won't be enforced by Swift Lint.
+* Colons always have no space on the left and one space on the right. Exceptions are the ternary operator `? :`, empty dictionary `[:]` and `#selector` syntax `addTarget(_:action:)`.
 
-### Logging
+**Preferred**:
+```swift
+class TestDatabase: Database {
+  var data: [String: CGFloat] = ["A": 1.2, "B": 3.2]
+}
+```
 
-We use [unified logging system](https://developer.apple.com/documentation/os/logging) to capture log messages. To send a message to the logging system, call the `os_log` function. Please use debug log level to protect users' privacy.
+**Not Preferred**:
+```swift
+class TestDatabase : Database {
+  var data :[String:CGFloat] = ["A" : 1.2, "B":3.2]
+}
+```
+
+* Long lines should be wrapped at around 70 characters. A hard limit is intentionally not specified.
+
+* Avoid trailing whitespaces at the ends of lines.
+
+* Add a single newline character at the end of each file.
+
+## Comments
+
+When they are needed, use comments to explain **why** a particular piece of code does something. Comments must be kept up-to-date or deleted.
+
+Avoid block comments inline with code, as the code should be as self-documenting as possible. _Exception: This does not apply to those comments used to generate documentation._
+
+Avoid the use of C-style comments (`/* ... */`). Prefer the use of double- or triple-slash.
+
+## Classes and Structures
+
+### Which one to use?
+
+Remember, structs have [value semantics](https://developer.apple.com/library/mac/documentation/Swift/Conceptual/Swift_Programming_Language/ClassesAndStructures.html#//apple_ref/doc/uid/TP40014097-CH13-XID_144). Use structs for things that do not have an identity. An array that contains [a, b, c] is really the same as another array that contains [a, b, c] and they are completely interchangeable. It doesn't matter whether you use the first array or the second, because they represent the exact same thing. That's why arrays are structs.
+
+Classes have [reference semantics](https://developer.apple.com/library/mac/documentation/Swift/Conceptual/Swift_Programming_Language/ClassesAndStructures.html#//apple_ref/doc/uid/TP40014097-CH13-XID_145). Use classes for things that do have an identity or a specific life cycle. You would model a person as a class because two person objects are two different things. Just because two people have the same name and birthdate, doesn't mean they are the same person. But the person's birthdate would be a struct because a date of 3 March 1950 is the same as any other date object for 3 March 1950. The date itself doesn't have an identity.
+
```

Comment:
is it forked from somewhere? do we need it?

Reply:
@jaceklyp Yep, forked from RW guide: https://github.com/duckduckgo/iOS/blob/02df76396be6400dcfadcb4ba7e390ce17e2cb60/styleguide/STYLEGUIDE.md?plain=1#L1-L2

Are you suggesting to remove the entire Classes and Structures section or just the "Which one to use"?